scripts/seurat_filter.R

```diff
@@ -36,31 +41,44 @@ if (is.null(opts$seurat_qc_rds) | is.null(opts$seurat_qc_rds) | is.null(opts$sou
     write("\n\nseurat_qc_rds, soupx_rds, and scdblfinder_tsv must be provided! One or more are missing!", stderr())
     quit(status=1)
 }
-output_name <- opts$output_name
+output_basename <- opts$output_basename
+seurat_raw_rds <- opts$seurat_raw_rds
 seurat_qc_rds <- opts$seurat_qc_rds
 soupx_rds <- opts$soupx_rds
 scdblfinder_tsv <- opts$scdblfinder_tsv
 
 write("\n\nLoading Files...", stderr())
+seurat_raw_obj <- readRDS(seurat_raw_rds)
 seurat_qc_obj <- readRDS(seurat_qc_rds)
 soupx_obj <- readRDS(soupx_rds)
 scDblFinder_df <- read.csv(scdblfinder_tsv, sep = "\t", header = TRUE, row.names=1)
-write("Done loading files!", stderr())
+write("Done loading files, getting aligned cell count", stderr())
+# Create and update a filter summary file
+filter_summary = paste0(output_basename, ".stats.tsv")
+write("Filter step\ttNumber of to remove\tNumber of cells", file=filter_summary)
+
+num_align <- ncol(seurat_raw_obj)
+write(paste("Alignment", "NA", num_align, sep="\t"), file=filter_summary, append = TRUE)
 
 write("Filtering Seurat Cells...", stderr())
-write("...Beginning SoupX Filtration...", stderr())
+write("...Beginning QC Filtration...", stderr())
+num_qc_filtered = ncol(seurat_qc_obj)
+write(paste("Seurat QC", (num_align - num_qc_filtered), num_qc_filtered, sep="\t"), file=filter_summary, append = TRUE)
 # Report mRNA
 filter_cells <- colnames(seurat_qc_obj)[which(!(colnames(seurat_qc_obj) %in% colnames(soupx_obj)))] # Get Seurat cells not found in SoupX
-if (length(filter_cells) == 0) {
+if (num_qc_filtered == 0) {
```

Comment:
I'm a little curious why this was changed. This `num_qc_filtered` seems like a worse approximation than filter cells.